itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/reexec/TestReExecuteKilledTezAMQueryPlugin.java

```diff
@@ -0,0 +1,207 @@
+        stmt.execute("dfs -put " + dataFilePath.toString() + " " + "kv1.txt");
+        stmt.execute("use " + testDbName);
+        stmt.execute("create table " + tblName + " (int_col int, value string) ");
+        stmt.execute("load data inpath 'kv1.txt' into table " + tblName);
+        stmt.execute("create function sleepMsUDF as '" + udfName + "'");
+        stmt.execute("grant select on table " + tblName + " to role public");
+
+        stmt.close();
+        conDefault.close();
+    }
+
+    @AfterClass
+    public static void afterTest() {
+        if (miniHS2 != null && miniHS2.isStarted()) {
+            miniHS2.stop();
+        }
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        hs2Conn = getConnection(miniHS2.getJdbcURL(), System.getProperty("user.name"), "bar");
+    }
+
+    public static Connection getConnection(String jdbcURL, String user, String pwd) throws SQLException {
+        Connection conn = DriverManager.getConnection(jdbcURL, user, pwd);
+        conn.createStatement().execute("set hive.support.concurrency = false");
+        return conn;
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        LlapBaseInputFormat.closeAll();
+        hs2Conn.close();
+    }
+
+
+
+    @Test
+    public void testKillQueryById() throws Exception {
+        String user = System.getProperty("user.name");
+        Connection con1 = getConnection(miniHS2.getJdbcURL(testDbName),
+                user, "bar");
+
+        final HiveStatement stmt = (HiveStatement)con1.createStatement();
+        final StringBuffer stmtQueryId = new StringBuffer();
+        ExceptionHolder originalQExHolder = new ExceptionHolder();
+        originalQExHolder.throwable = null;
+
+        // Thread executing the query
+        Thread tExecute = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    System.out.println("Executing query: ");
+                    stmt.execute("set hive.llap.execution.mode = none");
+
+                    // The test table has 500 rows, so total query time should be ~ 500*500ms
+                    stmt.executeAsync("select sleepMsUDF(t1.int_col, 100), t1.int_col, t2.int_col " +
+                            "from " + tableName + " t1 join " + tableName + " t2 on t1.int_col = t2.int_col");
+                    stmtQueryId.append(stmt.getQueryId());
+                    stmt.getUpdateCount();
+                } catch (SQLException e) {
+                    originalQExHolder.throwable = e;
+                }
+            }
+        });
+
+        tExecute.start();
+
+        // wait for other thread to create the stmt handle
+        int count = 0;
+        while (++count <= 10) {
+                Thread.sleep(2000);
+                String queryId;
+                if (stmtQueryId.length() != 0) {
+                    queryId = stmtQueryId.toString();
+                } else {
+                    continue;
```

Comment:
Shall remove else part if we move rest of the code with break inside "if" block.